### PR TITLE
Clustering

### PR DESCRIPTION
**Pull request by reubenkw**
Passes test. Writing the test was almost as difficult as the algorithm. Removed forcing C++11 from the makefile. Hopefully that isn't actually needed for something else :). 

Added a TODO to generalize a 2D point and a 3D point in the future.

### CODE REVIEW

File: jetson/include/core/cluster.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifndef CLUSTER_H
+#define CLUSTER_H
+
+#include <vector>
+
+// TODO: refactor to have a 2D point and a 3D point in a seperate file
```
Comment (elinliu2) [inline jetson/include/core/cluster.h:6]: can you use the point class in `jetson/core/point` might need to generalize to 2D

File: jetson/src/core/cluster.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        }
+    }
+    
+    return clusters;
+}
+
+void test_clustering() {
+    vector<ClPoint> points{ClPoint{.1, .1}, ClPoint{.1, .2}, ClPoint{.2, .3}, ClPoint{0, 1}, ClPoint{0, .9}, ClPoint{1, 1}};
+    vector<vector<ClPoint>> expected{{ClPoint{1, 1}}, {ClPoint{0, .9}, ClPoint{0, 1}}, {ClPoint{.1, .1}, ClPoint{.1, .2}, ClPoint{.2, .3}}};
+    vector<vector<ClPoint>> recieved = cluster(points, 0.2);
```
Comment (elinliu2) [inline jetson/src/core/cluster.cpp:59]: how do you determine d for the flowers?

**Review by elinliu2 — APPROVED** [review]

File: jetson/include/core/cluster.h (hunk 1 above)
Comment (reubenkw) [reply jetson/include/core/cluster.h:6]: I'll do this as part of coordinate transform

File: jetson/src/core/cluster.cpp (hunk 2 above)
Comment (reubenkw) [reply jetson/src/core/cluster.cpp:59]: No idea. It'll have to be estimated based on some test samples. The pictures we have so far have the points grouped pretty well so it shouldn't be hard

**Pull request merged**